core/src/serve_repair.rs

```diff
@@ -168,6 +168,24 @@ struct ServeRepairStats {
     err_sig_verify: usize,
     err_unsigned: usize,
     err_id_mismatch: usize,
+    outgoing_bytes_requested: HashMap<Pubkey, /*bytes*/ u64>,
+}
+
+#[derive(Default)]
+struct RepairStakeStats {
+    staked_peer_count: usize,
+    unstaked_peer_count: usize,
+    staked_response_bytes: u64,
+    unstaked_response_bytes: u64,
+    max_staked_peer_response_bytes: u64,
+    max_unstaked_peer_response_bytes: u64,
+}
+
+impl ServeRepairStats {
+    fn record_outgoing_bytes(&mut self, id: &Pubkey, bytes: u64) {
+        let requested = self.outgoing_bytes_requested.entry(*id).or_default();
```

Comment:
Can we limit the size of this hash-map so that it can't grow unbounded?
```rust
let num_entries = self.outgoing_bytes_requested.len();
match self.outgoing_bytes_requested.entry(*id) {
  Entry::Occupied(entry) => {
     *entry.get_mut() += bytes;
  },
  Entry::Vacant(entry) => if num_entries < 1_000 {
      entry.insert(bytes);
  }
};
````

Reply:
done